src/server/scene/basic_surface.h

```diff
@@ -145,6 +145,9 @@ class BasicSurface : public Surface
     auto depth_layer() const -> MirDepthLayer override;
     void set_depth_layer(MirDepthLayer depth_layer) override;
 
+    optional_value<geometry::Rectangle> clip_area() const override;
```

Comment:
I think we want to move off `mir::optional_value` in favour of `std::experimental::optional` (and then `std::optional` when we can require C++17)

The miral bits are OK because consistency in API is more important, but we don't currently have any `mir::optional_value` uses here.

Reply:
Done. It was mostly a search and replace job but I had to do conversion between `mir::optional_value` and `std::experimental::optional` in `window_info.cpp`